src/vmm/src/cpu_config/x86_64/cpuid/intel/normalize.rs

```diff
@@ -233,11 +233,19 @@ impl super::IntelCpuid {
 
     /// Update brand string entry
     fn update_brand_string_entry(&mut self) -> Result<(), NormalizeCpuidError> {
+        /// We always use this brand string.
+        const DEFAULT_BRAND_STRING_BASE: &[u8; 28] = b"Intel(R) Xeon(R) Processor @";
         // Get host brand string.
         let host_brand_string: [u8; BRAND_STRING_LENGTH] = host_brand_string();
+        let mut zero_padded_default_brand_string: [u8; BRAND_STRING_LENGTH] =
+            [0; BRAND_STRING_LENGTH];
+        // In case of error we use the default string without the " @"
+        zero_padded_default_brand_string[..DEFAULT_BRAND_STRING_BASE.len() - 2]
+            .copy_from_slice(&DEFAULT_BRAND_STRING_BASE[..DEFAULT_BRAND_STRING_BASE.len() - 2]);
 
         let default_brand_string =
-            default_brand_string(host_brand_string).map_err(NormalizeCpuidError::GetBrandString)?;
+            default_brand_string(host_brand_string, DEFAULT_BRAND_STRING_BASE)
+                .unwrap_or(zero_padded_default_brand_string);
```

Comment:
I think a significantly simpler solution would also work here.
```rust
    /// Update brand string entry
    fn update_brand_string_entry(&mut self) -> Result<(), NormalizeCpuidError> {
        // Get host brand string.
        let host_brand_string: [u8; BRAND_STRING_LENGTH] = host_brand_string();

        // If our desired default brand string can be formed, update the brand string.
        if let Ok(brand_string) = default_brand_string(host_brand_string) {
            self.apply_brand_string(&brand_string).map_err(NormalizeCpuidError::ApplyBrandString)?;
        }

        Ok(())
    }
```
or if we still wanted to preserve the brand string name just without caring about the frequency.
```rust
    const DEFAULT_BRAND_STRING: &[u8: BRAND_STRING_LENGTH] = b"Intel(R) Xeon(R) Processor @\0\0\0\0\0\0\0";
    const DEFAULT_BRAND_STRING_BASE: &[u8] = DEFAULT_BRAND_STRING_BASE[..28];
    /// Update brand string entry
    fn update_brand_string_entry(&mut self) -> Result<(), NormalizeCpuidError> {


        // Get host brand string.
        let host_brand_string: [u8; BRAND_STRING_LENGTH] = host_brand_string();

        //  Update the brand string
        let brand_string = default_brand_string(host_brand_string).unwrap_or(DEFAULT_FULL_BRAND_STRING_BASE);
        self.apply_brand_string(&brand_string).map_err(NormalizeCpuidError::ApplyBrandString)?;

        Ok(())
    }
```
(the amount of `\0`s I put in `DEFAULT_BRAND_STRING` may be wrong).